plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgrePrivilege.java

```diff
@@ -163,9 +163,16 @@ public short getPermission(PostgrePrivilegeType privilegeType) {
 
     public void setPermission(PostgrePrivilegeType privilegeType, boolean permit) {
         for (ObjectPermission permission : permissions) {
-            if (permission.privilegeType == privilegeType) {
+            if (permission.privilegeType != privilegeType) {
```

Comment:
Is this change intended? This will bloat `permissions` with multiple `privilegeType`-s.

Also it updates `permissions` array while iterating through it which is bad.

Reply:
it's intended. Before this implementation the permission array was overwritten, so alway when the userr change role and goes back, dbeaver would only check the last privilege checkbox, but let all the previous grant privilege statements on the permissionMap (so when user clicks on save button it shows all grant statement queries correct)

Reply:
i had wrote a message before asking your opinion about using another object to store permissions list just for use with check UI instead to change the current permission object, but i don't know why the message missed.